docs/concepts/how-ipfs-works.md

```diff
@@ -5,86 +5,171 @@ description: Learn how the InterPlanetary File System (IPFS) works and why it's
+IPFS nodes use Bitswap, a <VueCustomTooltip label="Unlike a request-response protocol, all nodes in the system receive every message transmitted, and decide whether the message received should be immediately discarded, stored or processed." underlined multiline is-medium>message-based protocol</VueCustomTooltip>, to fetch and send blocks of data. The process is as follows:
 
-To build a Merkle DAG representation of your content, IPFS often first splits it into _blocks_. Splitting it into blocks means that different parts of the file can come from different sources and be authenticated quickly. (If you've ever used BitTorrent, you may have noticed that when you download a file, BitTorrent can fetch it from multiple peers at once; this is the same idea.)
+1. Requesting nodes broadcast a <VueCustomTooltip label="A list of the CIDs of data blocks that an IPFS node wants to receive from peers in the network." underlined>wantlist</VueCustomTooltip> message to peers in the network.
+1. Peers receiving the wantlist eventually process and respond to the requester with one of the following:
+   - A _have_ message, indicating the peer has a CID requested.
+   - A _don't-have_ message, indicating the peer DOES NOT have a requested CID.
+1. The requester sends a _want-block_ message to peers that responded with a _have_ message, which requests that the peer send the block of data specified by the requesters wantlist.
+1. Peers send the block to the requester.
+1. As soon as the requesting node receives the wanted block, it broadcasts a message to peers that it no longer needs the requested block.
 
-::: tip
-It's easy to see a Merkle DAG representation of a file of your choice using the [DAG Builder visualizer](https://dag.ipfs.io/).
+:::callout
+Peers also store wantlists, so that if a peer receives requested blocks at a later time, it can then send them to the node that originally requested the data blocks. 
 :::
 
-Merkle DAGs are a bit of a ["turtles all the way down"](https://ipfs.io/ipfs/QmXoypizjW3WknFiJnKLwHCnL72vedxjQkDDP1mXWo6uco/wiki/Turtles_all_the_way_down.html) scenario; that is, _everything_ has a CID. Let's say you have a file, and its CID identifies it. What if that file is in a folder with several other files? Those files will have CIDs too. What about that folder's CID? It would be a hash of the CIDs from the files underneath (i.e., the folder's content). In turn, those files are made up of blocks, and each of those blocks has a CID. You can see how a file system on your computer could be represented as a DAG. You can also see, hopefully, how Merkle DAG graphs start to form. For a visual exploration of this concept, take a look at the [IPLD Explorer](https://explore.ipld.io/#/explore/QmSnuWmxptJZdLJpKRarxBMS2Ju2oANVrgbr2xWbie9b2D).
+### mDNS
```

Comment:
I wouldn't get into all the details of mDNS, we can link out or move it to another docs page.

The key thing to get across is how mDNS works in IPFS, and as far as I understand, it's to discover other peers in the local network.